### PR TITLE
test: Add tests for pool handling conflict txs

### PR DESCRIPTION
**Pull request by keroro520**

### CODE REVIEW

File: test/src/specs/tx_pool/conflict.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        node.generate_blocks(window.farthest() as usize + 2);
+
+        let (txa, txb) = conflict_transactions(node);
+        node.submit_transaction(&txa);
+
+        node.submit_block(&propose(node, &[&txa]));
+        (0..window.closest()).for_each(|_| {
+            node.submit_block(&blank(node));
+        });
+        node.submit_transaction(&txb);
```
Comment (u2) [inline test/src/specs/tx_pool/conflict.rs]: I think the submit tx will return an error result here.

Comment (keroro520) [reply test/src/specs/tx_pool/conflict.rs]: Updated. PTAL.

**Comment by keroro520** [comment]
> Why not any assertion after `generate_blocks`?

Assert committed transactions? Or what do you suggest to assert?
IMO, `generate_blocks` already includes that miner pick transactions into new blocks and node verify whether the new blocks are valid, which means if `generate_blocks` do not panic, then the blockchain runs ok.

**Comment by jjyr** [comment]
bors r=quake, jjy

**Comment by jjyr** [comment]
cherry pick to https://github.com/nervosnetwork/ckb/pull/1736

**Pull request closed without merging**